### PR TITLE
[vm] More logging

### PR DESCRIPTION
**Pull request by tnowacki**
- Logged the function for `expect_only_successful_execution`
- Logged for unverified modules loaded on chain

## Motivation

- logging is good

## Test Plan

- cargo test
- not sure how we normally test logging (if at all)

### CODE REVIEW

**Comment by dariorussi** [comment]
@metajack @sunshowers we are adding `libra-logger` to the` move-vm`. That adds a dependency.
I remember Rain kept `libra-logger` out of the `move-vm` in the past, but I don't remember the reason. I think it was for clean dependencies, and this should be ok, but it would be good if you could confirm

File: language/libra-vm/src/errors.rs [hunk 1]
```diff
@@ -132,16 +132,21 @@ pub fn convert_write_set_prologue_error(status: VMStatus) -> VMStatus {
...
 
-        status => {
-            crit!(
-                "[libra_vm] Unexpected error from known move function. Error: {:?}",
-                status
-            );
-            VMStatus::Error(StatusCode::UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION)
+            status => {
+                crit!(
+                    "[libra_vm] Unexpected error from known move function, '{}'. Error: {:?}",
```
Comment (bob-wilson) [inline language/libra-vm/src/errors.rs:144]: Nit (not new to this change): "move" should start with an uppercase letter, i.e., "...known Move function"

File: language/vm/src/errors.rs [hunk 2]
```diff
@@ -107,7 +108,11 @@ impl VMError {
     pub fn expect_no_verification_errors(self) -> Self {
         match self.status_type() {
             status_type @ StatusType::Deserialization | status_type @ StatusType::Verification => {
-                let message = format!("Unexpected verifier/deserialization error! This likely means there is code stored on chain is unverifiable!\nPrevious error {:?}", &self);
+                let message = format!(
+                    "Unexpected verifier/deserialization error! This likely means there is code \
```
Comment (bob-wilson) [inline language/vm/src/errors.rs:112]: Grammar (not new to this change): that last sentence has two verbs. I think it needs a "that": e.g., "...code stored on chain *that* is unverifiable!"

**Comment by sunshowers** [comment]
Yeah it wasn't obvious whether libra-logger should have been a dependency of the move runtime or not. I think eventually when we split up the Move VM entirely, we'll want to probably have logging/metrics interfaces.

**Comment by vgao1996** [comment]
@dariorussi I think it's fine we depend on libra-logger for now. One day if we do achieve a total Libra/Move separation, I'd assume us defining a logger trait for the VM which will get implemented for libra-logger or some adapters, but that a story for the far future...

**Comment by tnowacki** [comment]
- transaction-builder depends on the stdlib crate, which depends on the VM crate
- Having the logging in `vm/src/errors.rs` caused `transaction-builder` to not build on PowerPC32, which it needs to for various reasons 
- I moved the logging into vm/runtime which fixed it

**Pull request closed without merging**